workflow/controller/controller.go

```diff
@@ -1053,42 +1017,62 @@ func (wfc *WorkflowController) newPodInformer(ctx context.Context) cache.SharedI
 }
 
 func (wfc *WorkflowController) newConfigMapInformer() cache.SharedIndexInformer {
-	indexInformer := v1.NewFilteredConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
-		indexes.ConfigMapLabelsIndex: indexes.ConfigMapIndexFunc,
-	}, func(opts *metav1.ListOptions) {
-		opts.LabelSelector = common.LabelKeyConfigMapType
+
+	indexInformer := v1.NewConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
+		cache.NamespaceIndex: cache.MetaNamespaceIndexFunc,
 	})
-	log.WithField("executorPlugins", wfc.executorPlugins != nil).Info("Plugins")
-	if wfc.executorPlugins != nil {
-		indexInformer.AddEventHandler(cache.FilteringResourceEventHandler{
-			FilterFunc: func(obj interface{}) bool {
-				cm, err := meta.Accessor(obj)
-				if err != nil {
-					return false
+
+	isPluginCM := true
+	ctx := context.Background()
+
+	indexInformer.AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: func(obj interface{}) bool {
+			cm, err := meta.Accessor(obj)
+			if err != nil {
+				log.WithError(err).
+					Error("failed to get configmap")
+
+				return false
+			}
+
+			if cm.GetName() == wfc.configController.GetName() {
+				isPluginCM = false
+			} else {
+				isPluginCM = true
+				log.WithField("executorPlugins", wfc.executorPlugins != nil).
+					Info("Plugins")
+			}
+
+			return (cm.GetName() == wfc.configController.GetName()) ||
+				(cm.GetLabels()[common.LabelKeyConfigMapType] == common.LabelValueTypeConfigMapExecutorPlugin)
```

Comment:
so there is a _third_ type of ConfigMap that isn't handled here: the semaphore config update

Reply:
Sorry for asking, but I would like to clarify, `notifySemaphoreConfigUpdate` should be called for all CM caught by this informer or there is some exception (in the [current code](https://github.com/argoproj/argo-workflows/blob/master/workflow/controller/controller.go#L422) I think so)?

Reply:
the current code is all CMs in the managed namespace. I don't think there's a specific label for them right now (but maybe there should be)